packages/keyring-api/src/api/transaction.ts

```diff
@@ -0,0 +1,303 @@
+import type { Infer } from '@metamask/superstruct';
+import { array, enums, nullable, number, string } from '@metamask/superstruct';
+import { CaipChainIdStruct } from '@metamask/utils';
+
+import { AssetStruct } from './asset';
+import type { InferEquals } from '../superstruct';
+import { object } from '../superstruct';
+import type { Paginated } from '../utils';
+import { UuidStruct } from '../utils';
+
+/**
+ * This struct represents a participant in a transaction.
+ *
+ * @example
+ * ```ts
+ * {
+ *   address: '0x1234...',
+ *   asset: {
+ *     fungible: true,
+ *     type: 'eip155:1/slip44:60',
+ *     unit: 'ETH',
+ *     amount: '0.01',
+ *   },
+ * },
+ * ```
+ *
+ * @example
+ * ```ts
+ * {
+ *   address: '0x1234...',
+ *   asset: {
+ *     fungible: false,
+ *     id: 'eip155:1/erc721:0x06012c8cf97BEaD5deAe237070F9587f8E7A266d/771769',
+ *   },
+ * },
+ * ```
```

Comment:
Since it's `nullable` should we also have an example for this?

```suggestion
 * ```
 *
 * @example
 * ```ts
 * {
 *   address: '0x1234...',
 *   asset: null,
 * },
```